SimPPS/PPSPixelDigiProducer/plugins/PPSPixelDigiProducer.cc

```diff
@@ -0,0 +1,264 @@
+#ifndef SimPPS_PPSPixelDigiProducer_PPSPixelDigiProducer_h
+#define SimPPS_PPSPixelDigiProducer_PPSPixelDigiProducer_h
+
+// -*- C++ -*-
+//
+// Package:    PPSPixelDigiProducer
+// Class:      CTPPSPixelDigiProducer
+// 
+/**\class CTPPSPixelDigiProducer PPSPixelDigiProducer.cc SimPPS/PPSPixelDigiProducer/plugins/PPSPixelDigiProducer.cc
+
+ Description: <one line class summary>
+
+ Implementation:
+     <Notes on implementation>
+*/
+//
+// Original Author:  F.Ferro
+//
+
+#include "boost/shared_ptr.hpp"
+
+// system include files
+#include <memory>
+#include <vector>
+#include <map>
+#include <string>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+
+//  ****  CTPPS
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigi.h"
+#include "DataFormats/CTPPSDigi/interface/CTPPSPixelDigiCollection.h"
+
+#include "SimPPS/PPSPixelDigiProducer/interface/RPixDetDigitizer.h"
+#include "SimDataFormats/CrossingFrame/interface/MixCollection.h"
+
+#include "DataFormats/Common/interface/DetSet.h"
+
+// DB
+#include "CondFormats/DataRecord/interface/CTPPSPixelDAQMappingRcd.h"
+#include "CondFormats/DataRecord/interface/CTPPSPixelAnalysisMaskRcd.h"
+#include "CondFormats/CTPPSReadoutObjects/interface/CTPPSPixelDAQMapping.h"
+#include "CondFormats/CTPPSReadoutObjects/interface/CTPPSPixelAnalysisMask.h"
+#include "CondFormats/CTPPSReadoutObjects/interface/CTPPSPixelGainCalibrations.h"
+#include "RecoCTPPS/PixelLocal/interface/CTPPSPixelGainCalibrationDBService.h"
+
+// user include files
+#include "SimDataFormats/TrackingHit/interface/PSimHitContainer.h"
+#include "SimDataFormats/TrackingHit/interface/PSimHit.h"
+#include "SimDataFormats/CrossingFrame/interface/CrossingFrame.h"
+#include "SimDataFormats/CrossingFrame/interface/MixCollection.h"
+#include <cstdlib> // I need it for random numbers
+//needed for the geometry:
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+
+//Random Number
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/Utilities/interface/RandomNumberGenerator.h"
+#include "FWCore/Utilities/interface/Exception.h"
+#include "CLHEP/Random/RandomEngine.h"
+
+namespace CLHEP {
+  class HepRandomEngine;
+}
+
+class CTPPSPixelDigiProducer : public edm::EDProducer {
```

Comment:
This should inherit from one of the threaded EDProducers, likely `stream` is the simplest one.